### PR TITLE
Keyboard support for palette entries

### LINKED ISSUE #503: Implement scalable component library
__What should we do?__

- [ ] As a form designer, I expect the most commonly used components to appear at the top, so I don't need to scroll to use them.
- [ ] I want to search for components by name.
- [ ] I want to browse all available components.
- [ ] I expect components to be categorized by use case (such as "Basic Input", "Selection", "Presentation", "Action"...).
- [ ] (Stretch) I want to browse and add components to the end of the form using my keyboard.
- [ ] (Stretch) I want to have different UI break points (screen size, playground mode)
- [ ] [Integrate into `form-playground`](https://github.com/camunda/form-playground/issues/49)

image

__Why should we do it?__

We want to help form designers build forms in less time by providing component library that allows to easily discover, browse, and use form components, as well as to understand what these components are used for.

* Check out the [product hub epic](https://github.com/camunda/product-hub/issues/728) to gather more rationale.
* [Competitor analysis](https://miro.com/app/board/uXjVP8rk-qI=/)
* [Mockups](https://www.figma.com/file/CiTSN1vNETjIAy6W9jUIFp/Form-Editor?node-id=723%3A43296&t=T4tVSiG5XZRNoZJa-0)
* [Kick off Miro board](https://miro.com/app/board/uXjVPx8pfPU=/)

Related to https://github.com/camunda/product-hub/issues/728

### LINKED ISSUE #536: I want to browse and add components to the end of the form using my keyboard
### Is your feature request related to a problem? Please describe.

I want to browse and add components to the end of the form using my keyboard. I have to use my mouse currently via drag and drop as keyboard is not supported. This blocks keyboard users to add new components to the form.

### Describe the solution you'd like

Make browsing and adding form field components fully supported by keyboard.

Part of this was already sketched via https://github.com/bpmn-io/form-js/issues/503#issuecomment-1398213967 (branch: https://github.com/bpmn-io/form-js/tree/503-library-keyboard-support).

Kapture 2023-01-20 at 11 47 33

### Describe alternatives you've considered

/

### Additional context

Related to https://github.com/bpmn-io/form-js/issues/173
Related to https://github.com/bpmn-io/form-js/issues/503#issuecomment-1398213967

### PR DESCRIPTION
**Pull request by pinussilvestrus**
This makes the palette entries actual buttons so they can be controlled via keyboard. Additionally, this adds an `Enter` shortcut to the entries so one can add new form fields to the bottom of the form.

Closes #536

Kapture 2023-10-10 at 14 40 20

- [ ] This PR adds a new `form-js` element or visually changes an existing component.
  * => In that case, we need to ensure we follow up on this, e.g. by [creating an issue in Tasklist](https://github.com/camunda/tasklist/issues/new/choose)

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
The screencast is a bit misleading (focus border), seems like my screencast app is having some graphic troubles. It should work as expected in the demo.

**Review by vsgoulart — APPROVED** [review]
works well for me

does it maybe make sense to add support in this PR to delete items from the form when focuses and the user presses delete?

File: packages/form-js-editor/src/features/palette/components/PaletteEntry.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      }
+      <span class="fjs-palette-field-text">{ label }</span>
+    </button>
+  );
+}
+
+
+// helpers ///////////
+
+function getIndefiniteArticle(type) {
```
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply packages/form-js-editor/src/features/palette/components/PaletteEntry.js:48]: I just copied it over, it's inside for a longer time now

**Comment by pinussilvestrus** [comment]
For the delete via keyboard we had another PR that was blocked by a core issue: https://github.com/bpmn-io/form-js/pull/438. I have to double-check whether this affects this PR as well. I will test this one with the Desktop Modeler.

**Comment by vsgoulart** [comment]
@pinussilvestrus the visual regression tests failed, were the changes intentional?

**Comment by pinussilvestrus** [comment]
Should not, I will have a look

**Comment by pinussilvestrus** [comment]
> For the delete via keyboard we had another PR that was blocked by a core issue: #438. I have to double-check whether this affects this PR as well. I will test this one with the Desktop Modeler.

Works as expected 👍 I will handle the deletion shortcut separately still, as the PR is already open. Good candidate for spring cleaning as well.

Kapture 2023-10-12 at 09 18 54

**Comment by pinussilvestrus** [comment]
@volodymyr-melnykc @vsgoulart The failing visual regression test is because the icons are 1px moved compared to the previous version. 

I tried a few things to mitigate that, but it didn't work for all browsers reliably. I could take more time to fix this, but honestly, I'm not sure whether it's worth the time, as it doesn't look better or worse the other way around, in my opinion.

Let me know what you think; if we are fine with the 1px change, I will update the screenshots to make the tests green. 

Kapture 2023-10-12 at 11 00 32

**Comment by volodymyr-melnykc** [comment]
@pinussilvestrus It's ok for me! And yes, the new state (when the diff-line is moved to the right) looks more centered.

**Pull request merged**